pkg/asset/agent/image/ignition.go

```diff
@@ -380,9 +386,38 @@ func addExtraManifests(config *igntypes.Config, extraManifests *manifests.ExtraM
 	})
 
 	for _, file := range extraManifests.FileList {
-		extraFile := ignition.FileFromBytes(filepath.Join(extraManifestPath, filepath.Base(file.Filename)), user, mode, file.Data)
-		config.Storage.Files = append(config.Storage.Files, extraFile)
+
+		r := bytes.NewReader(file.Data)
+		decoder := yaml.NewDecoder(r)
+
+		var manifest map[string]interface{}
+		for n := 0; ; n++ {
+
```

Comment:
Why the empty line here?

Reply:
Exit condition(s) are defined below (either the decoder stops finding another document, or an error during decoding/marshaling happens)